packages/amplify-util-uibuilder/src/__tests__/getAmplifyDataSchema.test.ts

```diff
@@ -1,135 +1,83 @@
-      })),
-    }));
-  });
-
   beforeEach(() => {
     context = {
-      exeInfo: {
-        projectConfig: {
-          javascript: {
-            config: {
-              SourceDir: 'src',
-            },
-          },
-        },
-      },
       amplify: {
-        invokePluginMethod: () => ({}),
-      },
-      parameters: {
-        argv: [],
+        invokePluginMethod: mockedGetModelIntrospection,
       },
-      input: {
-        options: {
-          appId: 'testAppId',
-          envName: 'testEnvName',
-        },
-      },
-      print: { warning: jest.fn() },
     };
 
-    mockedFeatureFlags.getNumber.mockReturnValue(2);
-    getMockedSchema.mockClear();
-    awsMock.AmplifyBackend = jest.fn().mockReturnValue({
-      getBackendAPIModels: jest.fn().mockReturnValue({
-        promise: () => ({
-          Models: getMockedSchema(),
-        }),
-      }),
-    });
+    mockedGetModelIntrospection.mockReset();
   });
 
   it('should getAmplifyBackendModels', async () => {
-    const client = await AmplifyStudioClient.setClientInfo(context);
-    const dataSchema = await getAmplifyDataSchema(client);
-    expect(getMockedSchema).toBeCalled();
+    mockedGetModelIntrospection.mockReturnValue(sampleIntrospectionSchema);
+
+    const dataSchema = await getAmplifyDataSchema(context);
     expect(dataSchema).toBeDefined();
     expect(Object.keys(dataSchema!.models)).toContain('Blog');
   });
 
-  it('should not getAmplifyBackendModels and return undefined if graphQLTransformer version not supported', async () => {
-    mockedFeatureFlags.getNumber.mockReturnValue(1);
-    const client = await AmplifyStudioClient.setClientInfo(context);
-    const dataSchema = await getAmplifyDataSchema(client);
-    expect(getMockedSchema).not.toBeCalled();
-    expect(dataSchema).toBeUndefined();
-  });
+  it('should handle schema not found', async () => {
+    mockedGetModelIntrospection.mockReturnValue(undefined);
 
-  it('should handle Models not found', async () => {
-    awsMock.AmplifyBackend = jest.fn().mockReturnValue({
-      getBackendAPIModels: jest.fn().mockReturnValue({
-        promise: () => ({
-          Models: undefined,
-        }),
-      }),
-    });
-    const client = await AmplifyStudioClient.setClientInfo(context);
-    const dataSchema = await getAmplifyDataSchema(client);
+    const dataSchema = await getAmplifyDataSchema(context);
     expect(dataSchema).toBeUndefined();
-    expect(printerMock.debug).toHaveBeenCalledWith('Provided ResourceName: MyResourceName did not yield Models.');
+    expect(printerMock.debug).toHaveBeenCalledWith('Local schema not found');
   });
 
-  it('should handle network error', async () => {
-    awsMock.AmplifyBackend = jest.fn().mockReturnValue({
-      getBackendAPIModels: jest.fn().mockReturnValue({
-        promise: jest.fn().mockRejectedValue(new Error('Network Error')),
-      }),
+  it('should handle error', async () => {
+    const error = new Error('Something went wrong');
```

Comment:
How did we go from network error to general error? It seems like we're losing information with this change?

Reply:
The introspection api uses the local schema in the project and we don't need to make a http request anymore to the AmplifyBackend service to get the customer's schema.